content/blog/jetstream-java-client-03-consume.md

```diff
@@ -0,0 +1,131 @@
+So if the consumer stops then restarts, it will automatically restart where it left off and the configuration used to initialize
+the consumer will be remembered. A durable consumer is required when making a pull subscription and is optional
+when making a push subscription.
+
+An ephemeral consumer will only be tracked by the server while it is active and showing interest in the stream.
+Using consumer configuration options, an ephemeral can be manually set to start at a specific sequence or time.
+Ephemeral consumers are only allowed in push subscriptions.
+
+## Configuration Object
+The `ConsumerConfiguration` object has a builder to simplify setting options. Here is the builder skeleton showing all the builder methods.
+
+```
+ConsumerConfiguration c = ConsumerConfiguration.builder()
+    .durable(...)
+    .deliverPolicy(...)
+    .startSequence(...)
+    .startTime(...)
+    .deliverSubject(...)
+    .ackPolicy(...)
+    .ackWait(...)
+    .maxAckPending(...)
+    .replayPolicy(...)
+    .maxDeliver(...)
+    .filterSubject(...)
+    .rateLimit(...)
+    .sampleFrequency(...)
+    .idleHeartbeat(...)
+    .flowControl(...)
+    .build();
+```
+
+### Durable (Name)
+
+There is no default for the durable name. For push consumers, this is the only way to set the durable consumer name.
+For pull consumers, you can set the durable name in the `ConsumerConfiguration` object or in the `PullSubscribeOptions` object.
+
+### Deliver Policy / Start Sequence / Start Time
+
+When a consumer is first created, it can specify where in the stream it wants to start receiving messages.
+This is the `DeliverPolicy` and it's options are as follows:
+
+| Option  | Description  |
+| --- | --- |
+| All | All is the default policy. The consumer will start receiving from the earliest available message. |
+| Last | The consumer will start receiving messages with the last message. /
+| New | The consumer will only start receiving messages that were created after the consumer was created. /
+| ByStartSequence | The consumer is required to specify the `.startSequence(...)`, the sequence number to start on. It will receive the closest available sequence if that message was removed based on the stream limit policy. | 
+| ByStartTime | The consumer is required to specify the `.startTime(...)`, the time in the stream to start at. It will receive the closest available message on or after that time. | 
+
+### Deliver Subject
+
+Deliver subject essentially creates an alias core NATS subject for the stream. This means a core NATS subscriber could be set up to receive messages on the deliver subject,
+starting wherever the deliver policy was configured for the consumer. You could use a core NATS subscriber to access the stream by its original subject, but that would always start at
+the next message that appears on the subject, instead of where the consumer configured start sequence or start time. This is only allowed for push subscriptions.
```

Comment:
We definitely should not recommend people subscribe with core nats on the stream ingest subject - if thats what this is saying?

Reply:
I'll follow up with you to discuss